### PR TITLE
[CI] publish infrastructure for "latest"

### PR DESCRIPTION
**Pull request by marcdumais-work**
Add necessary CI infrastructure for automated publishing of latest `tsp-typescript-client` package. Also, in preparation to using said infrastructure, make small improvements in package's `package.json` and `README`.

Note: at this time, job `GitHub release` is expected to fail.

### CODE REVIEW

**Comment by marcdumais-work** [comment]
The `yarn upgrade` caused some 3PP license check failures. I ran in review mode locally, to create missing IP tickets:

```
[main] INFO This content is either not correctly mapped by the system, or requires review.
[main] INFO A review is required for npm/npmjs/@babel/helpers/7.24.1.
[main] INFO A review request already exists https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13922 .
[main] INFO A review is required for npm/npmjs/@babel/highlight/7.24.2.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13941 .
[main] INFO A review is required for npm/npmjs/@babel/helper-string-parser/7.24.1.
[main] INFO A review request already exists https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13923 .
[main] INFO A review is required for npm/npmjs/@babel/core/7.24.3.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13942 .
[main] INFO A review is required for npm/npmjs/@babel/compat-data/7.24.1.
[main] INFO A review request already exists https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13915 .
[main] INFO A review is required for npm/npmjs/@babel/code-frame/7.24.2.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13943 .
[main] INFO A review is required for npm/npmjs/@babel/helper-module-imports/7.24.3.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13944 .
[main] INFO A review is required for npm/npmjs/@babel/traverse/7.24.1.
[main] INFO A review request already exists https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13926 .
[main] INFO A review is required for npm/npmjs/@babel/generator/7.24.1.
[main] INFO A review request already exists https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13919 .
[main] INFO A review is required for npm/npmjs/@babel/plugin-syntax-typescript/7.24.1.
[main] INFO A review request already exists https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13924 .

```

File: RELEASE [hunk 1]
```diff
@@ -0,0 +1,8 @@
+tag: v0.0.0
```
Comment (marcdumais-work) [inline RELEASE:1]: Tag added on second commit in the repo:
https://github.com/eclipse-cdt-cloud/tsp-typescript-client/releases/tag/v0.0.0

File: yarn.lock [hunk 2]
```diff
@@ -3,81 +3,71 @@
 
 
 "@ampproject/remapping@^2.2.0":
-  version "2.2.1"
-  resolved "https://registry.npmjs.org/@ampproject/remapping/-/remapping-2.2.1.tgz#99e8e11851128b8702cd57c33684f1d0f260b630"
-  integrity sha512-lFMjJTrFL3j7L9yBxwYfCq2k6qqwHyzuUl/XBnif78PWTJYyL/dfowQHWE3sp6U6ZzqWiiIZnpTMO96zhkjwtg==
+  version "2.3.0"
+  resolved "https://registry.yarnpkg.com/@ampproject/remapping/-/remapping-2.3.0.tgz#ed441b6fa600072520ce18b43d2c8cc8caecc7f4"
```
Comment (bhufmann) [inline yarn.lock:7]: I noticed that the registry change everywhere from registry.npmjs.org to registry.yarnpkg.com. Any reason for that? Should we revert?

**Review by bhufmann — COMMENTED** [review]
@marcdumais-work I added question to the yarn.lock. Otherwise, it looks good.

File: yarn.lock (hunk 2 above)
Comment (marcdumais-work) [reply yarn.lock]: Reverted. It had to do with deleting my ~/.yarnrc , that contains the config for the npm server to use. Without this config, it felt-back to the yarn alias. 

Doing the following reverted it:
> yarn config set registry https://registry.npmjs.org
yarn upgrade

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+          release_file: 'RELEASE'
+          # Actions that run using the auto-generated GitHub token are
+          # not allowed to trigger a new workflow run. In this case we want
+          # the tag created by actions-gh-release to re-trigger the main workflow
+          # and result in publishing the package to npm.
+          # The following scopes are required when creating the committer token:
+          #  - repo:status, repo_deployment, public_repo, read:org
+          # See here for more details:
+          # https://docs.github.com/en/actions/security-guides/automatic-token-authentication#using-the-github_token-in-a-workflow
+          token: ${{ secrets.GH_COMMITTER_TOKEN }}
```
Comment (marcdumais-work) [inline .github/workflows/release.yml:36]: I added this secret in this repo

**Review by bhufmann — APPROVED** [review]
Looks good to me. Thanks!

**Pull request merged**